syntax-analysis.py

```diff
@@ -12,39 +20,70 @@ def fileAnalysis(file):
     printKeywords = 0 # holds the number of times the print function is called
     while line is not None:
         if sub == 0:
-            if line.__contains__('def'):
+            if line.__contains__('def '): # WE WILL NEED TO ADD FUNCATIONALITY TO CHECK IF THERE IS NO SPACE THIS IS A TEMP FIX
                     if line.__contains__('(') and line.__contains__(')') and line.__contains__(':'):
                         sub +=1
+                        out.write(line)
                         line = opened.readline()
                     elif line.find('(') == -1: # this denotes the the first paraenthesis for the def function is missing
                         # since strings are immutatable have to make a new string with the missing piece and then overwrite that line of the file
                         line = line[:line.find('def') + 3] + '(' + line[line.find('def') + 3:]
                         out.write(line)
-                        opened.readline()
+                        line = opened.readline()
                     elif line.find(')') == -1:
                         line = line[:line.find(':')] + ')' + line[line.find(':'):]
                         out.write(line)
-                        opened.readline()
+                        line = opened.readline()
                     else:
```

Comment:
if the line is missing both '(' and ')' then this else case will never be met and the first condition will be met instead.